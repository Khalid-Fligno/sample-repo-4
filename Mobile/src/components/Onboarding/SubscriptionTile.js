import React from "react";
import PropTypes from "prop-types";
import { View, Text, TouchableOpacity, StyleSheet } from 'react-native';
import SavingsBadge from "./SavingsBadge";
import fonts from "../../styles/fonts";
import colors from "../../styles/colors";
import RoundTick from "../../../assets/icons/RoundTick";
import EmptyRoundTick from "../../../assets/icons/EmptyRoundTick";

const SubscriptionTile = (props) => {
  const {
    title,
    price,
    priceNumber,
    onPress,
    comparisonPrice,
    selected,
    term,
    trialPeriod,
    savingsPercent,
    additionalText
  } = props;
  return (
    <TouchableOpacity
      onPress={onPress}
      style={[
        styles.subscriptionTile,
        {
          borderColor: selected ? colors.themeColor.color : colors.grey.dark,
        },
      ]}
    >
      <View style={styles.subscriptionTileContainer}>
        <View style={styles.textContainer}>
          <View>
            <View style={styles.subscriptionTitleContainer}>
              {<Text>{title} </Text>}
              <Text>{savingsPercent && <SavingsBadge text={`Save ${savingsPercent}%`}/>}</Text>
            </View>
            <Text
              style={[
                styles.priceTitle,
                {
                  color: selected ? colors.themeColor.color : colors.grey.dark,
                },
              ]}>
              {price}
              <Text style={styles.yearlySubTitle}>/ {term}

              </Text>
            </Text>
          </View>
          {comparisonPrice && (
            <Text style={styles.comparisonPrice}>{comparisonPrice}</Text>
          )}
          {savingsPercent && (
            <Text style={{ marginTop: 3 }}>
              <Text style={styles.subscriptionPriceText}>
                {`${price[0]}${(priceNumber / 12).toFixed(2)}`} / month
              </Text>
            </Text>
          )}
          {trialPeriod && (
            <Text style={[styles.subText, { marginTop: 1 }]}>
              {`After ${trialPeriod} free trial`}
            </Text>
          )}
          {additionalText && (
            <Text style={[styles.subText, { marginTop: 1 }]}>
              {additionalText}
            </Text>
          )}
        </View>
        <View style={styles.iconSubscriptionTileColumn}>
          {selected && <RoundTick height={50} width={60} />}
          {!selected && <EmptyRoundTick height={50} width={60} />}
        </View>
      </View>
    </TouchableOpacity>
  );
};

SubscriptionTile.propTypes = {
  title: PropTypes.string.isRequired,
  price: PropTypes.string.isRequired,
  term: PropTypes.string.isRequired,
  onPress: PropTypes.func.isRequired,
  savingsPercent: PropTypes.string,
  comparisonPrice: PropTypes.string,
  priceNumber: PropTypes.number.isRequired,
  isDiscounted: PropTypes.bool,
  selected: PropTypes.bool,
  currencyCode: PropTypes.any,
  currencySymbol: PropTypes.any,
};

SubscriptionTile.defaultProps = {
  savingsPercent: undefined,
  comparisonPrice: undefined,
  isDiscounted: false,
  selected: false,
};

const styles = StyleSheet.create({
  subscriptionTile: {
    justifyContent: "space-between",
    backgroundColor: "transparent",
    margin: 10,
    paddingHorizontal: 25,
    paddingVertical: 15,

    borderWidth: colors.themeColor.themeBorderWidth,
    borderColor: colors.grey.standard,
    borderRadius: 2,
  },
  subscriptionTilePrimary: {
    justifyContent: "space-between",
    backgroundColor: colors.coral.standard,
    margin: 10,
    padding: 15,
    paddingTop: 25,
    paddingBottom: 25,
    borderColor: colors.coral.standard,
    borderRadius: 2,
    shadowColor: colors.black,
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.4,
    shadowRadius: 3,
  },
  subscriptionTileContainer: {
    flexDirection: "row",
    justifyContent: "space-between",
  },
  iconSubscriptionTileColumn: {
    alignItems: "center",
    justifyContent: "center",
  },
  textContainer: {
    justifyContent: "center",
    flex: 1,
  },
  subscriptionTitleRow: {
    flexDirection: "row",
  },
  subscriptionTitleContainer: {
    flexDirection: "row",
    alignItems: "center",
    marginBottom: 3,
  },
  subscriptionTitleText: {
    letterSpacing: fonts.letterSpacing,
    fontFamily: fonts.GothamMedium,
    fontSize: 15,
    color: colors.charcoal.dark,
    marginBottom: 2,
  },
  yearlySubTitle: {
    fontFamily: fonts.GothamMedium,
    fontSize: 12,
    color: colors.grey.dark,
  },
  priceTitle: {
    fontSize: 30,
    fontFamily: fonts.bold,
  },
  comparisonPrice: {
    textDecorationLine: "line-through",
    textDecorationStyle: "solid",
    fontFamily: fonts.standard,
    fontSize: 12,
    color: colors.grey.light,
  },
  subscriptionPriceText: {
    fontFamily: fonts.GothamMedium,
    fontSize: 12,
    color: colors.grey.dark,
  },
  subText: {
    fontFamily: fonts.GothamMedium,
    fontSize: 12,
    color: colors.grey.dark,
  },
});

export default SubscriptionTile;