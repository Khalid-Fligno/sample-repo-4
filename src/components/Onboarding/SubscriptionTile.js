import React from 'react';
import PropTypes from 'prop-types';
import { View, Text, TouchableOpacity, StyleSheet } from 'react-native';
import Icon from '../../components/Shared/Icon';
import fonts from '../../styles/fonts';
import colors from '../../styles/colors';

const fullPriceMap = {
  '$15.99': '$19.99',
  '$39.99': '$49.99',
  '$111.99': '$139.99',
};

export default class SubscriptionTile extends React.PureComponent {
  constructor(props) {
    super(props);
    this.state = {
    };
  }
  render() {
    const {
      title,
      price,
      onPress,
      active,
      foundation,
    } = this.props;
    return (
      <TouchableOpacity
        onPress={onPress}
        style={active ? styles.subscriptionTileActive : styles.subscriptionTileInactive}
      >
        <Text style={active ? styles.subscriptionTileHeaderActive : styles.subscriptionTileHeaderInactive}>
          {title}
        </Text>
        {
          foundation && (
          <Text style={active ? styles.subscriptionPriceTextActiveStrikethrough : styles.subscriptionPriceTextInactiveStrikethrough}>
            {fullPriceMap[price]}
          </Text>
        )
        }
        <Text style={active ? styles.subscriptionPriceTextActive : styles.subscriptionPriceTextInactive}>
          {price}
        </Text>
        <View style={active ? styles.tickCircleActive : styles.tickCircleInactive}>
          {
            active && (
              <Icon
                name="tick-heavy"
                size={25}
                color={active ? colors.charcoal.standard : colors.grey.standard}
              />
            )
          }
        </View>
      </TouchableOpacity>
    );
  }
}

SubscriptionTile.propTypes = {
  title: PropTypes.string.isRequired,
  price: PropTypes.string.isRequired,
  onPress: PropTypes.func.isRequired,
  active: PropTypes.bool.isRequired,
  foundation: PropTypes.bool,
};

SubscriptionTile.defaultProps = {
  foundation: false,
};

const styles = StyleSheet.create({
  subscriptionTileInactive: {
    flex: 1,
    alignItems: 'center',
    backgroundColor: colors.white,
    marginLeft: 5,
    marginRight: 5,
    paddingTop: 10,
    paddingBottom: 10,
    borderWidth: 2,
    borderColor: colors.grey.light,
    borderRadius: 4,
  },
  subscriptionTileActive: {
    flex: 1,
    alignItems: 'center',
    backgroundColor: colors.white,
    marginLeft: 5,
    marginRight: 5,
    paddingTop: 10,
    paddingBottom: 10,
    borderWidth: 2,
    borderColor: colors.charcoal.standard,
    borderRadius: 4,
    shadowColor: colors.charcoal.standard,
    shadowOpacity: 0.5,
    shadowOffset: { width: 0, height: 2 },
    shadowRadius: 3,
  },
  subscriptionTileHeaderInactive: {
    fontFamily: fonts.bold,
    fontSize: 14,
    textAlign: 'center',
    color: colors.grey.standard,
    marginBottom: 5,
  },
  subscriptionTileHeaderActive: {
    fontFamily: fonts.bold,
    fontSize: 14,
    textAlign: 'center',
    color: colors.charcoal.standard,
    marginBottom: 5,
  },
  subscriptionPriceTextInactiveStrikethrough: {
    fontFamily: fonts.standard,
    fontSize: 14,
    color: colors.grey.standard,
    textDecorationLine: 'line-through',
    textDecorationStyle: 'solid',
  },
  subscriptionPriceTextActiveStrikethrough: {
    fontFamily: fonts.standard,
    fontSize: 14,
    color: colors.charcoal.standard,
    textDecorationLine: 'line-through',
    textDecorationStyle: 'solid',
  },
  subscriptionPriceTextInactive: {
    fontFamily: fonts.bold,
    fontSize: 14,
    color: colors.grey.standard,
    marginBottom: 5,
  },
  subscriptionPriceTextActive: {
    fontFamily: fonts.bold,
    fontSize: 14,
    color: colors.charcoal.standard,
    marginBottom: 5,
  },
  tickCircleInactive: {
    alignItems: 'center',
    justifyContent: 'center',
    width: 50,
    height: 50,
    borderWidth: 5,
    borderColor: colors.grey.light,
    borderRadius: 25,
  },
  tickCircleActive: {
    alignItems: 'center',
    justifyContent: 'center',
    width: 50,
    height: 50,
    borderWidth: 5,
    borderColor: colors.coral.standard,
    borderRadius: 25,
  },
});
