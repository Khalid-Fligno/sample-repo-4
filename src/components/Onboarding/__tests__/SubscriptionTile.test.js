import React from 'react';
import renderer from 'react-test-renderer';
import SubscriptionTile from '../SubscriptionTile';

describe('New Recipe Badge', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<SubscriptionTile
      title="Test"
      price="Test"
      term="Test"
      onPress={jest.fn()}
      priceNumber={1}
    />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
