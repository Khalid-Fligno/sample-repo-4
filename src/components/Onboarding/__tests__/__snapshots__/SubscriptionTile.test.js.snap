// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New Recipe Badge renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "rgba(0, 0, 0, 0.4 )",
      "borderColor": "rgb(255, 255, 255)",
      "borderRadius": 2,
      "borderWidth": 1,
      "justifyContent": "space-between",
      "margin": 10,
      "opacity": 1,
      "padding": 15,
      "paddingBottom": 25,
      "paddingTop": 25,
      "shadowColor": "rgb(0, 0, 0)",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.4,
      "shadowRadius": 3,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "rgb(255, 255, 255)",
              "fontFamily": "GothamBold",
              "fontSize": 14,
              "marginBottom": 2,
            }
          }
        >
          Test
          <Text
            style={
              Object {
                "fontFamily": "GothamBook",
              }
            }
          >
            Test / Test
          </Text>
        </Text>
        <View
          style={
            Object {
              "backgroundColor": "rgb(238, 48, 66)",
              "borderRadius": 2,
              "marginBottom": 6,
              "marginLeft": 8,
              "padding": 3,
              "paddingBottom": 0,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "rgb(255, 255, 255)",
                "fontFamily": "GothamBold",
                "fontSize": 8,
              }
            }
          >
            SAVE 40%
          </Text>
        </View>
      </View>
      <Text>
        <Text
          style={
            Object {
              "color": "rgb(255, 255, 255)",
              "fontFamily": "GothamBold",
              "fontSize": 12,
            }
          }
        >
          $0.08
           / month 
        </Text>
        <Text
          style={
            Object {
              "color": "rgb(255, 255, 255)",
              "fontFamily": "GothamBook",
              "fontSize": 12,
            }
          }
        >
          billed annually
        </Text>
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Text />
    </View>
  </View>
</View>
`;
